subsys/nrf_security/src/drivers/nrf_cc3xx/Kconfig

```diff
@@ -1,262 +1,456 @@
+config PSA_NEED_CC3XX_SHA_1
 	bool
-	prompt "PSA SHA1 support - cc3xx" if !PSA_PROMPTLESS
-	default y if !PSA_DEFAULT_OFF
+	default y
+	select PSA_ACCEL_SHA1
 	depends on PSA_WANT_ALG_SHA_1
+	depends on PSA_USE_CC3XX_HASH_DRIVER
 
-config PSA_CRYPTO_DRIVER_ALG_SHA_224_CC3XX
+config PSA_NEED_CC3XX_SHA_224
 	bool
-	prompt "PSA SHA-224 support - cc3xx" if !PSA_PROMPTLESS
-	default y if !PSA_DEFAULT_OFF
+	default y
+	select PSA_ACCEL_SHA224
 	depends on PSA_WANT_ALG_SHA_224
+	depends on PSA_USE_CC3XX_HASH_DRIVER
 
-config PSA_CRYPTO_DRIVER_ALG_SHA_256_CC3XX
+config PSA_NEED_CC3XX_SHA_256
 	bool
-	prompt "PSA SHA-256 support - cc3xx" if !PSA_PROMPTLESS
-	default y if !PSA_DEFAULT_OFF
+	default y
+	select PSA_ACCEL_SHA256
 	depends on PSA_WANT_ALG_SHA_256
+	depends on PSA_USE_CC3XX_HASH_DRIVER
 
-config PSA_CRYPTO_DRIVER_ALG_STREAM_CIPHER_CC3XX
+config PSA_NEED_CC3XX_HASH_DRIVER
 	bool
-	prompt "PSA stream cipher support - cc3xx" if !PSA_PROMPTLESS
-	default y if !PSA_DEFAULT_OFF
-	depends on PSA_WANT_ALG_STREAM_CIPHER
+	default y
+	depends on PSA_NEED_CC3XX_SHA_1   || \
+			   PSA_NEED_CC3XX_SHA_224 || \
+			   PSA_NEED_CC3XX_SHA_256
+
+# CC3xx Key Generation Driver
 
-config PSA_CRYPTO_DRIVER_ECC_BRAINPOOL_P_R1_256_CC3XX
+config PSA_NEED_CC3XX_RSA_KEY_SIZE_1024
 	bool
-	prompt "PSA ECC Brainpool256r1 support - cc3xx" if !PSA_PROMPTLESS
-	default y if !PSA_DEFAULT_OFF
-	depends on PSA_WANT_ECC_BRAINPOOL_P_R1_256
+	default y
+	depends on PSA_WANT_RSA_KEY_SIZE_1024
+	depends on PSA_USE_CC3XX_KEY_PAIR_DRIVER
+
+config PSA_NEED_CC3XX_RSA_KEY_SIZE_1536
+	bool
+	default y
+	depends on PSA_WANT_RSA_KEY_SIZE_1536
+	depends on PSA_USE_CC3XX_KEY_PAIR_DRIVER
+
+config PSA_NEED_CC3XX_RSA_KEY_SIZE_2048
+	bool
+	default y
+	depends on PSA_WANT_RSA_KEY_SIZE_2048
+	depends on PSA_USE_CC3XX_KEY_PAIR_DRIVER
+
+config PSA_NEED_CC3XX_RSA_KEY_SIZE_3072
+	bool
+	default y
+	depends on PSA_WANT_RSA_KEY_SIZE_3072
+	depends on PSA_USE_CC3XX_KEY_PAIR_DRIVER
+	depends on HAS_HW_NRF_CC312
+
+config PSA_NEED_CC3XX_KEY_PAIR_RSA
+	bool
+	default y
+	depends on PSA_NEED_CC3XX_RSA_KEY_SIZE_1024 || PSA_NEED_CC3XX_RSA_KEY_SIZE_1536 || \
+			   PSA_NEED_CC3XX_RSA_KEY_SIZE_2048 || PSA_NEED_CC3XX_RSA_KEY_SIZE_3072
+
+config PSA_NEED_CC3XX_KEY_PAIR_P224
+	bool
+	default y
+	select PSA_ACCEL_KEY_PAIR_P224
+	depends on PSA_WANT_ECC_SECP_R1_224
+	depends on PSA_USE_CC3XX_KEY_PAIR_DRIVER
+
+config PSA_NEED_CC3XX_KEY_PAIR_P256
+	bool
+	default y
+	select PSA_ACCEL_KEY_PAIR_P256
+	depends on PSA_WANT_ECC_SECP_R1_256
+	depends on PSA_USE_CC3XX_KEY_PAIR_DRIVER
 
-config PSA_CRYPTO_DRIVER_ECC_MONTGOMERY_255_CC3XX
+config PSA_NEED_CC3XX_KEY_PAIR_P384
 	bool
-	prompt "PSA ECC Curve25519 support - cc3xx" if !PSA_PROMPTLESS
-	default y if !PSA_DEFAULT_OFF
+	default y
+	select PSA_ACCEL_KEY_PAIR_P384
+	depends on PSA_WANT_ECC_SECP_R1_384
+	depends on PSA_USE_CC3XX_KEY_PAIR_DRIVER
+
+config PSA_NEED_CC3XX_KEY_PAIR_P521
+	bool
+	default y
+	# TODO: Add PSA_ACCEL_KEY_P521 when the symbol is available
```

Comment:
I am missing the KEY_PAIR indeed. But I could find the other reference that you are talking about, maybe you are looking at an old revision? I remember that I had to fix all the non existent options since it broke the build.

Reply:
Probably, the TODO should be fixed then.